docs/misc/config.md

```diff
@@ -0,0 +1,95 @@
+Configuration
+=============
+
+Configurations define settings and consumer preferences that can be applied to a mod instance. Forge uses a configuration system using [TOML][toml] files and read with [night-config][nightconfig].
+
+Creating a Configuration
+------------------------
+
+A configuration can be created using a subtype of `IConfigSpec`. Forge implements the type via `ForgeConfigSpec` and enables its construction through `ForgeConfigSpec#Builder`. The builder can separate the config values into sections via `Builder#push` to create a section and `Builder#pop` to leave a section. Afterwards, the configuration can be built using one of two methods:
```

Comment:
Do we want to expose the `IConfigSpec` type in the docs? It may be worth simplifying down to just the existence of `ForgeConfigSpec`.

Reply:
Technically, Forge exposes `IConfigSpec` as a result of `ModLoadingContext#registerConfig`. That's why I breifly mention the configuration interface. Although, that's all the documentation mentions as it more specifically uses `ForgeConfigSpec`.